addons/account/models/account_move.py

```diff
@@ -872,26 +873,25 @@ def _compute_name(self):
                 continue
 
             move_has_name = move.name and move.name != '/'
-            if move_has_name or move.state != 'posted':
-                if not move.posted_before and not move._sequence_matches_date():
-                    if move._get_last_sequence():
-                        # The name does not match the date and the move is not the first in the period:
-                        # Reset to draft
-                        move.name = False
-                        continue
-                else:
-                    if move_has_name and move.posted_before or not move_has_name and move._get_last_sequence():
-                        # The move either
-                        # - has a name and was posted before, or
-                        # - doesn't have a name, but is not the first in the period
-                        # so we don't recompute the name
-                        continue
-            if move.date and (not move_has_name or not move._sequence_matches_date()):
+            if (move_has_name or move.state == 'draft') and not move.posted_before and not move._sequence_matches_date():
+                # The name does not match the date and the move is not the first in the period:
+                # Reset to draft
+                move.name = False
+                continue
+            if move.date and not move_has_name and move.state != 'draft':
                 move._set_next_sequence()
 
-        self.filtered(lambda m: not m.name and not move.quick_edit_mode).name = '/'
         self._inverse_name()
 
+    @api.depends('date', 'journal_id', 'move_type', 'name', 'posted_before', 'sequence_number', 'sequence_prefix', 'state')
+    def _compute_name_placeholder_form(self):
+        for move in self:
+            if (not move.name or not move.posted_before) and not move._get_last_sequence():
```

Comment:
The best example as to why it should be `or` and not `and` is if you post the first move of the sequence and then reset it to draft. In that case the placeholder has to be back because the sequence is not fixed anymore but the move has a name so, with the condition using `and`, it will have no placeholder and therefore if the user removes the name. Having no name and no placeholder, the field will be replaced by the span "Draft" making it impossible to edit the sequence template.

But it won't show a placeholder for all draft moves like you say since as from the second move, `not move._get_last_sequence()` is falsy so the `if` is not entered, no placeholder is computed and "Draft" is displayed instead.